unittests/ProfilerTest.cpp

```diff
@@ -0,0 +1,41 @@
+#include "util/Profiler.h"
+#include <gtest/gtest.h>
+#include <thread>
+
+namespace util::detail {
+TEST(ProfilerTest, HasReturnValue)
+{
+    auto [ret, time] = timed([]() {
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+        return 8;
+    });
+
+    ASSERT_EQ(ret, 8);
+    ASSERT_NE(time, 0);
+}
+
+TEST(ProfilerTest, ReturnVoid)
+{
+    auto time =
+        timed([]() { std::this_thread::sleep_for(std::chrono::seconds(1)); });
+
+    ASSERT_NE(time, 0);
+}
+
+struct FunctorTest
+{
+    void
+    operator()() const
+    {
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+};
+
+TEST(ProfilerTest, Functor)
+{
+    auto time = timed(FunctorTest());
+
+    ASSERT_NE(time, 0);
+}
+
+}  // namespace util::detail
```

Comment:
Github does not like it when we don't add the final `\n` to files. You can configure your IDE to automatically insert newline at end of files and it will usually work fine that way.